### PR TITLE
Add creating dict without file reading. Add challenge example

### PR DESCRIPTION
**Pull request by ObjatieGroba**
New features

 - Add functions to add attribute, value and vendor to exists rc_handle.
It is much more effective than `rc_read_dictionary_from_buffer`
 - Add challenge example that uses new functions

Continuation of #61

### CODE REVIEW

File: src/radchallenge.c [hunk 1]
```diff
@@ -0,0 +1,210 @@
+/*
+ * radchallenge.c - a sample c program showing how to embed the configuration of a radius and to challenge
+ * client, using the FreeRADIUS Client Library without an external configuration and dictionary file .
```
Comment (nmav) [inline src/radchallenge.c]: Here it mentions the freeradius client library, I think it should mention radcli instead

File: src/radchallenge.c [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+#include <radcli/radcli.h>
+#include <string.h>
+
+#define AUTH_PORT_PASSWORD ":1812:testing123"
+#define ACCT_PORT_PASSWORD ":1813:testing123"
+#define SERVER_ADDR "localhost"
+
+int
+main (int argc, char **argv)
+{
```
Comment (nmav) [inline src/radchallenge.c:46]: I think it makes sense to include it as a test in tests/ so that this functionality is tested.

**Review by nmav — COMMENTED** [review]
Thank you for this and sorry for the late response, I missed this. I have some minor comments as part of this review. I think if you can address them and squash the commits to a single (or two), I think it will be ready for merging.

File: src/radchallenge.c (hunk 2 above)
Comment (ObjatieGroba) [reply src/radchallenge.c:46]: Is it possible to run such test?
If yes, how can I do it?

I have no idea how tests should look like. I will try to add one with rc_dict_addattr

Comment (nmav) [reply src/radchallenge.c:46]: Yes these tests run when you commit in github via travis. You can see their run:
https://travis-ci.org/github/radcli/radcli/builds/719301652

Just check the scripts in tests/ directory to get a feeling of how it works (e.g., tcp-tests.sh), you can add more tests via Makefile.am

**Comment by nmav** [comment]
Could you also rebase it to master?

**Comment by nmav** [comment]
@ObjatieGroba this didn't compile as is and the commit message was kind of confusing (as well as contained reference to not a name, and an non-existing email). I've cleaned it up to compile on the CI, replaced strcpy() with strlcpy() and reformatted the message in #71. Could you check it whether it is ok with you, and suggest a more elaborate commit message?

**Pull request closed without merging**